.github/workflows/_build_docker.yaml

```diff
@@ -0,0 +1,55 @@
+# Copyright (c) 2022 - 2023, Oracle and/or its affiliates. All rights reserved.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/.
+
+# This reuseable workflow builds the Macaron Docker image.
+# If release-tag is provided. The final image will be tagged and push to ghcr.io/oracle-samples/macaron:release-tag
+
+name: Build and push Docker image
+on:
+  workflow_call:
+    inputs:
+      release-tag:
+        description: The release tag to push the Docker image
+        required: false
+        type: string
+
+env:
+  IMAGE_NAME: ghcr.io/oracle-samples/macaron
+
+jobs:
+  build-docker:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Check out repository
+      uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8 # v3.1.0
+      with:
+        fetch-depth: 0
+
+    - name: Download artifact
+      uses: actions/download-artifact@9782bd6a9848b53b110e712e20e42d89988822b7 # v3.0.1
+      with:
+        path: dist
+
+    - name: Log in to GitHub Container Registry
+      run: docker login ghcr.io --username ${{ github.actor }} --password ${{ secrets.GITHUB_TOKEN }}
+
+    - name: Build the Docker image
+      run: |
+        ./scripts/dev_scripts/build_docker.sh ${{ env.IMAGE_NAME }} ${{ github.workspace }} ${{ inputs.release-tag }}
+
+    - name: Test the built Docker image
+      env:
+        # This will be picked up by run_macaron.sh.
+        MACARON_IMAGE_TAG: test
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      run: |
+        ./scripts/dev_scripts/integration_tests_docker.sh ${{ github.workspace }} ./scripts/release_scripts/run_macaron.sh
+
+    - name: Push the built Docker image
+      run: |
+        if [[ -n "${{ inputs.release-tag }}" ]];
```

Comment:
You can use an `if` condition on the step [as shown here](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsif) to prevent the step from running.

Reply:
This part is removed in https://github.com/oracle-samples/macaron/commit/24da909332996f30775f68257a844b0d2b9cbb6a